src/v64.rs

```diff
@@ -54,4 +54,12 @@ define_casts!(
     (u16x4, i16x4, as_i16x4),
     (i16x4, u16x4, as_u16x4),
     (u8x8, i8x8, as_i8x8),
-    (i8x8, u8x8, as_u8x8));
+    (i8x8, u8x8, as_u8x8),
+    (i8x8, i16x8, as_i16x8),
+    (i16x4, i32x4, as_i32x4),
+    (i32x2, i64x2, as_i64x2),
+    (u8x8, u16x8, as_u16x8),
+    (u16x4, u32x4, as_u32x4),
+    (u32x2, u64x2, as_u64x2)
```

Comment:
Oh I thought these were only for sem-width casts? What are the semantics of the different-width casts? Does each lane just get promoted?

Reply:
> Does each lane just get promoted?

Yes, that's what `simd_cast` does and what I intended. @BurntSushi how did you envision this? The number of `as_...` functions is going to explode if I keep doing this for the arm intrinsics. I would be fine with it, but maybe we should consider providing `simd_cast` as part of the library API.